code-challenges.test.js

```diff
@@ -1,52 +1,109 @@
+        return code[lowerCaseChar] || char
+    })
+
+    return encodedChars.join("")
+}
+describe("encoded", () => {
+
+    it("should encode 'Lackadaisical' to 'L4ck4d41s1c4l'", () => {
+        const secretCodeWord1 = "Lackadaisical"
+        expect(encoded(secretCodeWord1)).toBe("L4ck4d41s1c4l")
+    })
+    it("should encode 'Gobbledygook' to 'G0bbl3dyg00k'", () => {
+        const secretCodeWord2 = "Gobbledygook"
+        expect(encoded(secretCodeWord2)).toBe("G0bbl3dyg00k")
+    })
+    it("should encode 'Eccentric' to '3cc3ntr1c'", () => {
+        const secretCodeWord3 = "Eccentric"
+        expect(encoded(secretCodeWord3)).toBe("3cc3ntr1c")
+    })
+})
+
+// --------------------2) 
+// Pseudo Code:
+// Input: an array of strings and a single letter
+// Output: an array of strings (all the words that have that particular letter)
+// - Use the filter method on the array of `words` to create a new array that only has the words that include the particular letter from the input
+// - Use the lower case method to convert the words and the letter to lowercase so that comparing it will make it case-insensitive
+// - If the word contains the letter, it is returned in the filtered array
+
+const filterByLetter = (words, letter) => {
+    return words.filter(word => word.toLowerCase().includes(letter.toLowerCase()))
+}
+
+describe("filterByLetter", () => {
+    const fruitArray = ["Mango", "Cherry", "Apricot", "Blueberry", "Peach", "Kiwi"]
+
+    it("should return ['Mango', 'Apricot', 'Peach'] if the letter 'a' is chosen", () => {
+        const filterLetterA = "a"
+        expect(filterByLetter(fruitArray, filterLetterA)).toEqual(["Mango", "Apricot", "Peach"])
+    })
+
+    it("should return ['Cherry', 'Blueberry', 'Peach'] if the letter 'e' is chosen", () => {
+        const filterLetterE = "e"
+        expect(filterByLetter(fruitArray, filterLetterE)).toEqual(["Cherry", "Blueberry", "Peach"])
+    })
+})
+
+
+// --------------------3) 
+// Pseudo Code:
+// Input: an array of 5 numbers
+// Output: a boolean value (true if the array is a full house, false if not)
+// -Use the length method to check if the array has exactly 5 numbers, otherwise it will return a string that states there isn't enough numbers
+// -If it has 5 numbers in the array, it will be sorted in ascending order using the sort method and the comparator function. (Putting it in ascending order makes it easier to compare the numbers to see if there is a match)
+// -Use conditionals to check if there is a full house in the array
+// -One conditional checks if the first three elements in the sorted array are equal (to represent three of a kind) and the last two must be equal (to represent the pair)
+// Used the logical OR to check if the other condition is met, which checks if the first two values are the same and the last three are the same 
+// If any of the conditions are met, the return should be true, otherwise it will be false
```

Comment:
This pseudo code does a great job covering every step and logic, but it can def be shortened and tidied up